tests/qos/test_buffer.py

```diff
@@ -1888,3 +1959,189 @@ def test_buffer_model_test(duthosts, rand_one_dut_hostname, conn_graph_facts):
+
+    # Determine gearbox_delay with platform name, so far only MSN3800 has gear_box installed
+    if duthost.facts["platform"] not in ["x86_64-mlnx_msn3800-r0"]:
+        gearbox_delay = 0
+    else:
+        gearbox_delay_keys = duthost.shell('redis-cli -n 6 keys PERIPHERAL_TABLE*')['stdout']
+        gearbox_delay = float(duthost.shell('redis-cli -n 6 hget "{}" "gearbox_delay"'.format(gearbox_delay_keys))['stdout'])
+
+    # Get cable length from config DB
+    # Command: redis-cli -n 4 hget "CABLE_LENGTH|AZURE"  'Ethernet0'
+    cable_length_keys = duthost.shell('redis-cli -n 4 keys *CABLE_LENGTH*')['stdout']
+    cable_length_raw = duthost.shell('redis-cli -n 4 hget "{}" "{}"'.format(cable_length_keys, port_to_test))['stdout']
+    if cable_length_raw and cable_length_raw.endswith('m'):
+        cable_length = float(cable_length_raw[:-1])
+    else:
+        logging.error("failed to get a valid cable length from config db for port {}".format(port_to_test))
+        return False, None
+
+    logging.info('port_speed = {}, port_mtu = {}, cable_length = {}'.format(port_speed, port_mtu, cable_length))
+
+    # Get port lanes number from config DB
+    # Command: redis-cli -n 4 hget "PORT|Ethernet0" 'lanes'
+    port_lanes = duthost.shell('redis-cli -n 4 hget "PORT|{}" "lanes"'.format(port_to_test))['stdout']
+    is_8lane = port_lanes and len(port_lanes.split(',')) == 8
+
+    if not ASIC_TABLE_KEYS_LOADED:
+        # Get cell size from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'cell_size'
+        asic_keys = duthost.shell('redis-cli -n 6 keys *ASIC_TABLE*')['stdout']
+        CELL_SIZE = float(duthost.shell('redis-cli -n 6 hget "{}" "cell_size"'.format(asic_keys))['stdout'])
+
+        # Get PIPELINE_LATENCY from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'pipeline_latency'
+        PIPELINE_LATENCY = float(duthost.shell('redis-cli -n 6 hget "{}" "pipeline_latency"'.format(asic_keys))['stdout']) * 1024
+
+        # Get MAC_PHY_DELAY from state DB
+        # Command: redis-cli -n 6 hget "ASIC_TABLE|MELLANOX-SPECTRUM-2" 'mac_phy_delay'
+        MAC_PHY_DELAY = float(duthost.shell('redis-cli -n 6 hget "{}" "mac_phy_delay"'.format(asic_keys))['stdout']) * 1024
+
+    if not LOSSLESS_TRAFFIC_PATTERN_KEYS_LOADED:
+        # Get LOSSLESS_MTU from config DB
+        # Command: redis-cli -n 4 hget 'LOSSLESS_TRAFFIC_PATTERN|AZURE' 'mtu'
+        lossless_traffic_keys = duthost.shell('redis-cli -n 4 keys LOSSLESS_TRAFFIC_PATTERN*')['stdout']
+        LOSSLESS_MTU = float(duthost.shell('redis-cli -n 4 hget "{}" "mtu"'.format(lossless_traffic_keys))['stdout'])
+
+        # Get SMALL_PACKET_PERCENTAGE from config DB
+        # Command: redis-cli -n 4 hget 'LOSSLESS_TRAFFIC_PATTERN|AZURE' 'small_packet_percentage'
+        SMALL_PACKET_PERCENTAGE = float(
+            duthost.shell('redis-cli -n 4 hget "{}" "small_packet_percentage"'.format(lossless_traffic_keys))['stdout'])
```

Comment:
same comment as above

Reply:
added and new function "get_lossless_traffic_pattern_data_from_db"